content/open-source/sigstore/policy-controller/disallowing-unsafe-sysctls-with-policy-controller.md

```diff
@@ -0,0 +1,191 @@
+
+Open a new file with `nano` or your preferred editor:
+
+```shell
+nano /tmp/cip.yaml
+```
+
+Copy the following policy to the `/tmp/cip.yaml` file:
+
+```
+apiVersion: policy.sigstore.dev/v1beta1
+kind: ClusterImagePolicy
+metadata:
+  name: unsafe-sysctls-mask-cue
+spec:
+  match:
+  - version: "v1"
+    resource: "pods"
+  images: [glob: '**']
+  authorities: [static: {action: pass}]
+  mode: enforce
+  policy:
+    includeSpec: true
+    type: "cue"
+    data: |
+      spec: {
+        securityContext:
+          sysctls: [...{
+            name: "kernel.shm_rmid_forced" |
+                  "net.ipv4.ip_local_port_range" |
+                  "net.ipv4.ip_unprivileged_port_start" |
+                  "net.ipv4.tcp_syncookies" |
+                  "net.ipv4.ping_group_range"
+          }]
+      }
+```
+
+This policy will ensure that any pod that has a sysctl defined in its spec will only be admitted if it matches a parameter from the list.
+
+Save the file and then apply the policy:
+
+```bash
+kubectl apply -f /tmp/cip.yaml
+```
+
+You will receive output showing the policy is created:
+
+```
+clusterimagepolicy.policy.sigstore.dev/unsafe-sysctls-mask-cue
+```
+
+Next you will test the policy with a failing pod spec. Once you have confirmed that the admission controller is rejecting pods using unsafe sysctls, you'll create a pod with a safe parameter and admit it into your cluster.
+
+## Step 3 — Testing the `ClusterImagePolicy`
+
+Now that you have a policy defined, you can test that it successfully rejects or accepts admission requests.
+
+Use `nano` or your preferred editor to create a new file `/tmp/pod.yaml` and copy in the following pod spec that uses an unsafe sysctl:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: yolo
+spec:
+  securityContext:
+    sysctls:
+    - name: kernel.msgmax
+      value: "65536"
+  containers:
+  - name: "app"
+    image: docker.io/ubuntu
+```
+
+Apply the pod spec and check for the Policy Controller admission denied message:
+
+```
+kubectl apply -f /tmp/pod.yaml
+```
+
+```
+Error from server (BadRequest): error when creating "/tmp/pod.yaml": admission webhook "policy.sigstore.dev" denied the request: validation failed: failed policy: unsafe-sysctls-mask-cue: spec.containers[0].image
+index.docker.io/library/ubuntu@sha256:854037bf6521e9c321c101c269272f756e481fb5f167ae032cb53da08aebcd5a failed evaluating cue policy for ClusterImagePolicy: failed to evaluate the policy with error: spec.securityContext.sysctls.0.name: 5 errors in empty disjunction: (and 5 more errors)
+```
+
+The first line show the error message and the failing `ClusterImagePolicy` name. The second line contains the image ID, along with the specific CUE error message showing the policy violation.
+
+Edit the `/tmp/pod.yaml` file and change the `sysctls` section to use the following safe parameter:
+
+```
```

Comment:
I think it may make it more obvious to keep the `sysctls` in here

```suggestion
    sysctls:
    - name: net.ipv4.tcp_syncookies
      value: "1"
```